ui/src/app/utils/runtime-utils.tsx

```diff
@@ -128,6 +129,19 @@ const compareMachineMemory = (oldRuntime: RuntimeConfig, newRuntime: RuntimeConf
   };
 };
 
+export const compareGpu = (oldRuntime: RuntimeConfig, newRuntime: RuntimeConfig): RuntimeDiff => {
+  const oldGpuExists = !!oldRuntime.gpuConfig;
+  const newGpuExists = !!newRuntime.gpuConfig;
+  return {
+    desc: 'Change GPU config',
+    previous: oldGpuExists ? `${oldRuntime.gpuConfig.numOfGpus} ${oldRuntime.gpuConfig.gpuType} GPU` : 'No GPUs',
+    new: newGpuExists ?  `${newRuntime.gpuConfig.numOfGpus} ${newRuntime.gpuConfig.gpuType} GPU` : 'No GPUs',
+    differenceType: (!oldGpuExists && !newGpuExists) || (oldGpuExists && newGpuExists)
```

Comment:
I think the second set of parens should just wrap all 3 remaining predicates - they are highly related